tests/test_data_sampler.py

```diff
@@ -128,3 +128,21 @@ def test_split_sampler_even_size(num_samples, num_parts):
         print(count)
     expected_count = int(num_samples + num_parts - 1) // num_parts * num_parts
     assert total_count == expected_count, (total_count, expected_count)
+
+
+@pytest.mark.parametrize('seq_lengths', [[np.random.randint(10, 100) for _ in range(N)],
+                                         [(np.random.randint(10, 100), np.random.randint(10, 100)) for _ in range(N)]])
+@pytest.mark.parametrize('max_num_tokens', [200, 500])
+@pytest.mark.parametrize('max_num_sentences', [-1, 5])
+@pytest.mark.parametrize('required_batch_size_multiple', [1, 5])
+@pytest.mark.parametrize('shuffle', [True, False])
+@pytest.mark.parametrize('seed', [100, None])
+def test_bounded_budget_sampler(seq_lengths, max_num_tokens, max_num_sentences,
+                                required_batch_size_multiple, shuffle, seed):
+    sampler = s.BoundedBudgetSampler(seq_lengths, max_num_tokens, max_num_sentences,
+                                     required_batch_size_multiple, shuffle, seed)
+    print(sampler)
+    total_sampled_ids = []
+    for batch_sample_ids in sampler:
+        total_sampled_ids.extend(batch_sample_ids)
```

Comment:
When shuffle is False, you may test that all indexes are covered.

Reply:
added 
```python
assert sorted(total_sampled_ids) == list(range(len(total_sampled_ids)))
```
in the test script